azure-pipelines-pr.yml

```diff
@@ -0,0 +1,90 @@
+variables:
+  # Cannot use key:value syntax in root defined variables
+  - name: _TeamName
+    value: DotNetCore
+  - name: _PublishUsingPipelines
+    value: true
+  - name: _DotNetArtifactsCategory
+    value: .NETCore
+
+pr:
+- main
+
+stages:
+- stage: build
+  dependsOn: []
+  displayName: Build
+  jobs:
+  - template: /eng/common/templates/jobs/jobs.yml
+    parameters:
+      enableMicrobuild: false
+      enablePublishBuildArtifacts: true
+      enablePublishTestResults: false
+      enablePublishBuildAssets: true
+      enablePublishUsingPipelines: ${{ variables._PublishUsingPipelines }}
+      enableTelemetry: true
+      helixRepo: dotnet/arcade-services
+      jobs:
+      - job: Windows_NT
+        timeoutInMinutes: 90
+        pool:
+          ${{ if eq(variables['System.TeamProject'], 'internal')}}:
+            name: NetCore1ESPool-Internal
+            demands: ImageOverride -equals 1es-windows-2019
+          ${{ if eq(variables['System.TeamProject'], 'public')}}:
+            name: NetCore-Public          
+            demands: ImageOverride -equals 1es-windows-2019-open
+
+        variables:
+        - _InternalBuildArgs: ''
+
+        strategy:
+          matrix:
+            # Only build debug in non-official builds
+            ${{ if or(eq(variables['System.TeamProject'], 'public'), in(variables['Build.Reason'], 'PullRequest')) }}:
+                debug_configuration:
+                  _BuildConfig: Debug
+                  _PublishType: none
+                  _SignType: test
+                  _DotNetPublishToBlobFeed : false
+            release_configuration:
+              _BuildConfig: Release
+              # PRs or external builds are not signed.
+              ${{ if or(eq(variables['System.TeamProject'], 'public'), in(variables['Build.Reason'], 'PullRequest')) }}:
+                _PublishType: none
+                _SignType: test
+                _DotNetPublishToBlobFeed : false
+
+        steps:
+        - checkout: self
+          clean: true
+
+        - task: UseDotNet@2
+          displayName: Install Correct .NET Version
+          inputs:
+            useGlobalJson: true
+
+        - task: UseDotNet@2
+          displayName: Install .NET Version 3.1
+          inputs:
+            version: 3.1.x
+
+        - task: NuGetToolInstaller@0
+          inputs:
+            versionSpec: 6.1.x
+
+        - powershell: eng/set-version-parameters.ps1
+          displayName: Calculate release version variables
+
+        - script: eng/common/cibuild.cmd
+            -configuration $(_BuildConfig)
+            -prepareMachine
+            $(_InternalBuildArgs)
+            /p:Test=false
+            /P:Sign=false
+          name: Build
+          displayName: Build / Publish
+          condition: succeeded()
+
+        - ${{ if in(variables['Build.Reason'], 'PullRequest') }}:
```

Comment:
if you decide to disallow manual builds of this pipeline entirely, this condition becomes unnecessary